### ISSUE TITLE
[Accessibility]: Code insight card clickable legend items are not accessible from keyboard

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#44055]
### Audit type

Keyboard navigation

### Problem description

In some insights (insights with more than one series), legend items are clickable; they turn on/off rendered series (lines) on the chart. At the moment, non of them are clickable from a keyboard. 

### Expected behavior

Legend items, when they are clickable and interactive, should be rendered with buttons (or at least implement button behaviours, focus ring and keyboard shortcuts). 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @joelkw @felixfbecker @vovakulikov

**Comment by vovakulikov** [comment]
This is P0 since it violates keyboard accessibility requirement.
